src/main/kotlin/com/terning/server/kotlin/application/scrap/ScrapService.kt

```diff
@@ -0,0 +1,45 @@
+package com.terning.server.kotlin.application
+
+import com.terning.server.kotlin.application.scrap.ScrapRequest
+import com.terning.server.kotlin.domain.internshipAnnouncement.InternshipAnnouncementRepository
+import com.terning.server.kotlin.domain.scrap.Scrap
+import com.terning.server.kotlin.domain.scrap.ScrapRepository
+import com.terning.server.kotlin.domain.scrap.exception.ScrapErrorCode
+import com.terning.server.kotlin.domain.scrap.exception.ScrapException
+import com.terning.server.kotlin.domain.scrap.vo.Color
+import com.terning.server.kotlin.domain.user.UserRepository
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+@Transactional(readOnly = false)
+class ScrapService(
+    private val scrapRepository: ScrapRepository,
+    private val userRepository: UserRepository,
+    private val internshipAnnouncementRepository: InternshipAnnouncementRepository,
+) {
+    @Transactional
+    fun scrap(
+        userId: Long,
+        internshipAnnouncementId: Long,
+        request: ScrapRequest,
+    ) {
+        if (scrapRepository.existsByInternshipAnnouncementIdAndUserId(userId, internshipAnnouncementId)) {
+            throw ScrapException(ScrapErrorCode.EXISTS_SCRAP_ALREADY)
+        }
+
+        val announcement =
+            internshipAnnouncementRepository.findById(internshipAnnouncementId)
+                .orElseThrow { ScrapException(ScrapErrorCode.INTERN_SHIP_ANNOUNCEMENT_NOT_FOUND) }
+
+        val user =
+            userRepository.findById(userId)
+                .orElseThrow { ScrapException(ScrapErrorCode.USER_NOT_FOUND) }
+
+        val color = Color.from(request.color)
+        val scrap = Scrap.of(user, announcement, color)
```

Comment:
아래처럼 named argument를 사용하면 코드 가독성이 좋아질 것 같아요! 어떻게 생각하시나요?
```suggestion
        val scrap = Scrap.of(
            user = user, 
            internshipAnnouncement = announcement, 
            color = color
        )
```

Reply:
WOWOWO... 이거군요! 😮

확실히 네이밍 아규먼트를 사용하니까 가독성이 훨씬 좋아지네요.
약간의 중복처럼 느껴질 수도 있지만, 실수를 줄일 수 있고, 특히 협업할 때 의도가 명확히 드러나서 더 좋은 것 같아요!
말씀 주신 방향대로 개선하겠습니다 :) 💪